### PR TITLE
Move the mail box testsuite to new ztest API

### PR DESCRIPTION
**Pull request by enjiamai**
Migrate following tests to new ztest API:

- tests/kernel/mbox/mbox_api
- tests/kernel/mbox/mbox_usage

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

File: tests/kernel/mbox/mbox_api/boards/qemu_x86_lakemont.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# Need a little bit more stack space
+# after moving to new ztest API
+CONFIG_TEST_EXTRA_STACK_SIZE=64
```
Comment (andyross) [inline tests/kernel/mbox/mbox_api/boards/qemu_x86_lakemont.conf:6]: Might consider just a slightly larger increment.  It's true that our test suite really likes small stacks, but if 64 more bytes is enough, then we're within 64 bytes of a known overflow on at least one platform!  Someone is going to have to come back to this test in the near future and adjust it again, it's almost guaranteed.

File: tests/kernel/mbox/mbox_api/src/test_mbox_api.c [hunk 2]
```diff
@@ -605,86 +604,102 @@ void test_define_multi_mbox(void)
...
 
-void test_mbox_multiple_waiting_get(void)
+ZTEST(mbox_api, test_mbox_multiple_waiting_get)
 {
 	info_type = MULTIPLE_WAITING_GET;
 	tmbox(&mbox);
+
+	/* cleanup the sender threads */
+	for (int i = 0; i < 5 ; i++) {
+		k_thread_abort(&waiting_get_tid[i]);
```
Comment (andyross) [inline tests/kernel/mbox/mbox_api/src/test_mbox_api.c:692]: One of the interesting side effects of ztest2 re-ordering tests is that it's forced us to fix a bunch of state/cleanup issues like this one with existing tests.  I actually was thinking this was a misfeature (it makes it hard to find "the test you are working on" in output when developing), but now I'm thinking maybe we want to go further and have a "CONFIG_ZTEST_RANDOMIZE_TEST_ORDER" or something for use in CI/stress environments to further harden the tests.  Something to consider, @yperess?

**Review by andyross — APPROVED** [review]
Looks great, a few random notes.

File: tests/kernel/mbox/mbox_api/src/test_mbox_api.c (hunk 2 above)
Comment (enjiamai) [reply tests/kernel/mbox/mbox_api/src/test_mbox_api.c:692]: Your concern is totally right again, @andyross . I think we have "CONFIG_ZTEST_SHUFFLE" in place to make test running in a random order. And your concern is totally right , this test cannot fully passing the testing every time with this config set. I will fix it ASAP. Thanks for pointing this out!

**Pull request merged**